### PR TITLE
add functionality to record all moves

### PR DESCRIPTION
**Pull request by anonchickenlegs**
image

New feature shows piece movements throughout game.  Movements are displayed under captured pieces section. mfer heads are displayed to the left of description of move

### CODE REVIEW

File: server/chess/chess.functions.js [hunk 1]
```diff
@@ -124,21 +127,24 @@ function suggestMove(game, turn) {
...
+        promotion: topMove.promotion,
+        piece: topMove.piece
     };
 }
 
 
 
 function processMove(game, move, playerId) {
     const player = game.players.find(p => p.id === playerId);
+    player.moves.push(move)
```
Comment (nngmi) [inline server/chess/chess.functions.js:139]: should this happen AFTER the move is confirmed and the checks are all done?  What if the mover goes and puts in an invalid move?  will it make it into the move histroy

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 node_modules
 client/build/**
+client/.env
```
Comment (nngmi) [inline .gitignore:3]: remove these before chec kin ?

File: client/.env [hunk 3]
```diff
@@ -0,0 +1 @@
+REACT_APP_SERVER_URL=http://localhost:3000
```
Comment (nngmi) [inline client/.env]: remove this before checkin?

File: server/chess/chess.functions.js [hunk 4]
```diff
@@ -167,6 +172,7 @@ function processMove(game, move, playerId) {
     game.board = board;
     game.castling = castling;
     game.moves.push(move);
+    player.moves.push(move)
```
Comment (nngmi) [inline server/chess/chess.functions.js:175]: i see. so we now keep duplicative moves.  `game.moves` (interleaved) between two players and `player.moves`.  I am generally okay with that.

Comment (nngmi) [reply server/chess/chess.functions.js:175]: I could see us ripping out `game.moves` in future PR.

**Review by nngmi — APPROVED** [review]
LG.

Assume you have:

1. Run `npm test`.  I don't know if the addition of `piece` as a processMove requirement will require adding `piece` a lot everywhere.

2. Run some quick local testing games assume you did.

File: .gitignore (hunk 2 above)
Comment (anonchickenlegs) [reply .gitignore:3]: sorry not sure what you mean. I put it here so that people don't commit their local .env

**Pull request merged**